### ISSUE TITLE
TCP Retransmission when a process takes more than 360 seconds on GKE

### ISSUE THREAD

**Issue by emilwallner** [kubernetes/kubernetes#91954]
**What happened**:
I make a POST request to a Flask app wrapped in Gunicorn that makes a 360 seconds delay. Then, the app returns the response and sends a TCP package [FIN, ACK] to close the connection with the client. However, this response is lost between the Pod and the NodePort. The client never gets the response and the [FIN, ACK] package. The app is stuck sending a TCP Retransmission. I make the request through the external IP of the Node, and the port I've exposed via a NodePort service. 

**What you expected to happen**:
I expected the POST request to work for more than 360 seconds. I can't find any error messages nor documentation that explains this behavior. 

**How to reproduce it (as minimally and precisely as possible)**:
Create a flask app with Gunicorn that accepts a POST request. When it receives the request make a 360 seconds time delay before it responds. Expose the Pod with a NodePort. 

**Anything else we need to know?**:
If I use a ClusterIP instead of the NodePort - it works. I don't run into the 360 seconds time limit. Furthermore, the POST request through the NodePort works if the delay is under 300 seconds. 

**Environment**:
I'm using Kubernetes(1.17.5-gke.9), Gunicorn (20.0.4), and Python 3.7. I'm using the keep alive and timeout flags for Gunicorn.

**Comment by athenabot** [comment]
/triage unresolved

Comment `/remove-triage unresolved` when the issue is assessed and confirmed.

🤖 I am a bot run by vllry. 👩‍🔬

**Comment by athenabot** [comment]
@robscott
If this issue has been triaged, please comment `/remove-triage unresolved`.

If you aren't able to handle this issue, consider unassigning yourself and/or adding the `help-wanted` label.

🤖 I am a bot run by vllry. 👩‍🔬

**Comment by robscott** [comment]
Hey @emilwallner, thanks for the bug report! I've spent some time digging through the code and unfortunately haven't found anything quite yet. I've got a few follow up questions for you that could help debug this:

1. Did this issue appear for you in GKE 1.17? Have you run into this on older Kubernetes versions?
2. Was there any other network activity during this test?
3. Have you run into this issue with any non-POST requests?  
4. Does this still happen if you use a load balancer with [network endpoint groups](https://cloud.google.com/kubernetes-engine/docs/how-to/standalone-neg)?
